test_stop_meas_cont_acquisition_3 test randomly fails on travis-ci

This can be observed in this [build](https://travis-ci.com/reszelaz/sardana/jobs/230121128).

```console
======================================================================
FAIL: test_stop_meas_cont_acquisition_3 (sardana.tango.pool.test.test_measurementgroup.TangoAcquisitionTestCase)
Stop of the synchronized acquisition with two channels from the same controller using hardware trigger
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/usr/local/lib/python3.5/dist-packages/sardana-3.0.0a0-py3.5.egg/sardana/tango/pool/test/test_measurementgroup.py", line 253, in stop_meas_cont_acquisition
    self.assertEqual(state, desired_state, msg)
AssertionError: tango._tango.DevState.FAULT != tango._tango.DevState.ON : channel tango://sardana-test:10000/expchan/_test_ct_ctrl_1/1 state after stop is FAULT (should be ON)
----------------------------------------------------------------------
Ran 307 tests in 833.453s
```

--- a/src/sardana/pool/poolcontrollers/DummyCounterTimerController.py
+++ b/src/sardana/pool/poolcontrollers/DummyCounterTimerController.py
@@ -139,7 +139,7 @@ class DummyCounterTimerController(CounterTimerController):
         if self._armed:
             sta = State.Moving
             status = "Armed"
-        elif axis in self.counting_channels:
+        elif axis in self.counting_channels and self.start_time is not None:
             channel = self.channels[idx]
             now = time.time()
             elapsed_time = now - self.start_time
@@ -182,7 +182,7 @@ class DummyCounterTimerController(CounterTimerController):
         if self._armed:
             return  # still armed - no trigger/gate arrived yet
         # if in acquisition then calculate the values to return
-        if self.counting_channels:
+        if self.counting_channels and self.start_time is not None:
             now = time.time()
             elapsed_time = now - self.start_time
             for axis, channel in list(self.read_channels.items()):
@@ -263,12 +263,16 @@ class DummyCounterTimerController(CounterTimerController):
                                      AcqSynch.HardwareGate):
             self._disconnect_hardware_synchronization()
             self._armed = False
+        self.start_time = None
 
     def AbortOne(self, axis):
         if axis not in self.counting_channels:
             return
         now = time.time()
-        elapsed_time = now - self.start_time
+        if self.start_time is not None:
+            elapsed_time = now - self.start_time
+        else:
+            elapsed_time = 0
         self._finish(elapsed_time, axis)
 
     def GetCtrlPar(self, par):

--- a/src/sardana/pool/poolcontrollers/DummyTwoDController.py
+++ b/src/sardana/pool/poolcontrollers/DummyTwoDController.py
@@ -264,7 +264,7 @@ class BasicDummyTwoDController(TwoDController):
         if self._armed:
             sta = State.Moving
             status = "Armed"
-        elif axis in self.counting_channels:
+        elif axis in self.counting_channels and self.start_time is not None:
             channel = self.channels[idx]
             now = time.time()
             elapsed_time = now - self.start_time
@@ -342,12 +342,16 @@ class BasicDummyTwoDController(TwoDController):
                                      AcqSynch.HardwareGate):
             self._disconnect_hardware_synchronization()
             self._armed = False
+        self.start_time = None
 
     def AbortOne(self, axis):
         if axis not in self.counting_channels:
             return
         now = time.time()
-        elapsed_time = now - self.start_time
+        if self.start_time is not None:
+            elapsed_time = now - self.start_time
+        else:
+            elapsed_time = 0
         self._finish(elapsed_time, axis)
 
     def getAmplitude(self, axis):

